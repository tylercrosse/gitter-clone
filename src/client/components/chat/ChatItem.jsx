import React from 'react';
import moment from 'moment';

const ChatItem = ({ username, text, createdAt }) => {
  const dateString = moment(createdAt).format('MMM MM HH:mm');
  return (
    <article className="chat-item">
      <div className="chat-item-container">
        <div className="chat-item-aside">
          <img
            className="avatar"
            src={'http://i.pravatar.cc/30?u=' + username} alt={username}
          />
        </div>
        <div className="chat-item-content">
          <div className="chat-item-details">
            <div className="chat-item-user">{username}</div>
            <div className="chat-item-time">{dateString}</div>
          </div>
          <div className="chat-item-text">{text}</div>
        </div>
      </div>
    </article>
  );
};

export default ChatItem;
