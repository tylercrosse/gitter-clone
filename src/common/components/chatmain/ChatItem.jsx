import React, { PropTypes } from 'react';
import moment from 'moment';

export const ChatItem = ({
  User,
  text,
  rawMarkup,
  createdAt,
  burstStart
}) => {
  const dateString = moment(createdAt).format('MMM DD HH:mm');
  if (burstStart) {
    return (
      <article className="chat-item burst-start">
        <div className="chat-item-container">
          <div className="chat-item-aside">
            <img
            className="avatar"
            src={'http://i.pravatar.cc/30?u=' + User.name} alt={User.name}
            />
          </div>
          <div className="chat-item-content">
            <div className="chat-item-details">
              <div className="chat-item-user">{User.name}</div>
              <div className="chat-item-time">{dateString}</div>
            </div>
            {rawMarkup ?
              <div
              className="chat-item-text"
              dangerouslySetInnerHTML={{__html: rawMarkup}}
              /> :
              <div className="chat-item-text">{text}</div>
            }
          </div>
        </div>
      </article>
    );
  }
  return (
    <article className="chat-item burst-continued">
      <div className="chat-item-container">
        <div className="chat-item-content">
          {rawMarkup ?
            <div
            className="chat-item-text"
            dangerouslySetInnerHTML={{__html: rawMarkup}}
            /> :
            <div className="chat-item-text">{text}</div>
          }
        </div>
      </div>
    </article>
  );
};

ChatItem.propTypes = {
  User: PropTypes.object.isRequired,
  text: PropTypes.string.isRequired,
  rawMarkup: PropTypes.string.isRequired,
  createdAt: PropTypes.string.isRequired,
  burstStart: PropTypes.bool.isRequired,
};

export default ChatItem;
