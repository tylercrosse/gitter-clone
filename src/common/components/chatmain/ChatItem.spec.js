import React from 'react';
// import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ChatItem from './ChatItem';

const setup = (propOverrides) => {
  const props = Object.assign({
    id: 0,
    burstStart: true,
    text: 'Use Redux',
    username: 'Dan',
    updatedAt: '2017-01-25T00:58:33.702Z',
    createdAt: '2017-01-25T00:58:33.702Z'
  }, propOverrides);
  const component = <ChatItem {...props} />;

  return {
    props,
    component
  };
};

describe('<ChatItem />', () => {
  describe('burstStart == true', () => {
    it('should render correctly without markup', () => {
      const { props } = setup();
      const renderedComponent = renderer.create(<ChatItem {...props} />);
      const tree = renderedComponent.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render correctly with markup', () => {
      const { props } = setup({rawMarkup: '<p>woohoo!</p>'});
      const renderedComponent = renderer.create(<ChatItem {...props} />);
      const tree = renderedComponent.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('burstStart == false', () => {
    it('should render correctly without markup', () => {
      const { props } = setup({burstStart: false});
      const renderedComponent = renderer.create(<ChatItem {...props} />);
      const tree = renderedComponent.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render correctly with markup', () => {
      const { props } = setup({
        burstStart: false,
        rawMarkup: '<p>woohoo!</p>'
      });
      const renderedComponent = renderer.create(<ChatItem {...props} />);
      const tree = renderedComponent.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
