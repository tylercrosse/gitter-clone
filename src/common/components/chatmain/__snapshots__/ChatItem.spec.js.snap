// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatItem /> burstStart == false should render correctly with markup 1`] = `
<article
  className="chat-item burst-continued"
>
  <div
    className="chat-item-container"
  >
    <div
      className="chat-item-content"
    >
      <div
        className="chat-item-text"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>woohoo!</p>",
          }
        }
      />
    </div>
  </div>
</article>
`;

exports[`<ChatItem /> burstStart == false should render correctly without markup 1`] = `
<article
  className="chat-item burst-continued"
>
  <div
    className="chat-item-container"
  >
    <div
      className="chat-item-content"
    >
      <div
        className="chat-item-text"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Use Redux</p>",
          }
        }
      />
    </div>
  </div>
</article>
`;

exports[`<ChatItem /> burstStart == true should render correctly with markup 1`] = `
<article
  className="chat-item burst-start"
>
  <div
    className="chat-item-container"
  >
    <div
      className="chat-item-aside"
    >
      <img
        alt="Dan"
        className="avatar"
        src="http://i.pravatar.cc/30?u=Dan"
      />
    </div>
    <div
      className="chat-item-content"
    >
      <div
        className="chat-item-details"
      >
        <div
          className="chat-item-user"
        >
          Dan
        </div>
        <div
          className="chat-item-time"
        >
          Jan 24 16:58
        </div>
      </div>
      <div
        className="chat-item-text"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>woohoo!</p>",
          }
        }
      />
    </div>
  </div>
</article>
`;

exports[`<ChatItem /> burstStart == true should render correctly without markup 1`] = `
<article
  className="chat-item burst-start"
>
  <div
    className="chat-item-container"
  >
    <div
      className="chat-item-aside"
    >
      <img
        alt="Dan"
        className="avatar"
        src="http://i.pravatar.cc/30?u=Dan"
      />
    </div>
    <div
      className="chat-item-content"
    >
      <div
        className="chat-item-details"
      >
        <div
          className="chat-item-user"
        >
          Dan
        </div>
        <div
          className="chat-item-time"
        >
          Jan 24 16:58
        </div>
      </div>
      <div
        className="chat-item-text"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Use Redux</p>",
          }
        }
      />
    </div>
  </div>
</article>
`;
